consumers/writers/timescale/init.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package timescale
+
+import (
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq" // required for SQL access
+	migrate "github.com/rubenv/sql-migrate"
+)
+
+// Config defines the options that are used when connecting to a TimescaleSQL instance
+type Config struct {
+	Host        string
+	Port        string
+	User        string
+	Pass        string
+	Name        string
+	SSLMode     string
+	SSLCert     string
+	SSLKey      string
+	SSLRootCert string
+}
+
+// Connect creates a connection to the TimescaleSQL instance and applies any
+// unapplied database migrations. A non-nil error is returned to indicate
+// failure.
+func Connect(cfg Config) (*sqlx.DB, error) {
+	url := fmt.Sprintf("host=%s port=%s user=%s dbname=%s password=%s sslmode=%s sslcert=%s sslkey=%s sslrootcert=%s", cfg.Host, cfg.Port, cfg.User, cfg.Name, cfg.Pass, cfg.SSLMode, cfg.SSLCert, cfg.SSLKey, cfg.SSLRootCert)
+
+	db, err := sqlx.Open("postgres", url)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := migrateDB(db); err != nil {
+		return nil, err
+	}
+
+	return db, nil
+}
+
+func migrateDB(db *sqlx.DB) error {
+	migrations := &migrate.MemoryMigrationSource{
+		Migrations: []*migrate.Migration{
+			{
+				Id: "messages_1",
+				Up: []string{
+					`CREATE TABLE IF NOT EXISTS messages (
+						time BIGINT NOT NULL,
+                        channel       UUID,
+                        subtopic      VARCHAR(254),
+                        publisher     UUID,
+                        protocol      TEXT,
+                        name          VARCHAR(254),
+                        unit          TEXT,
+                        value         FLOAT,
+                        string_value  TEXT,
+                        bool_value    BOOL,
+                        data_value    BYTEA,
+                        sum           FLOAT,
+                        update_time   FLOAT,
+                        PRIMARY KEY (time, publisher, subtopic, name)
+                    );
+					
+					SELECT create_hypertable('messages', 'time', create_default_indexes => FALSE, chunk_time_interval => 86400000, if_not_exists => TRUE);
```

Comment:
Please check the indentation. Why don't we create default indexes here? I would assume that the user will want indexes on the time column, and that's why it's TRUE by default (please have in mind that I have never used TimescaleDB, this is only according to the official docs).

Reply:
It works more like the InfluxDB implementation in that respect than the postgres implementation.  There's no ID column and there might be multiple messages at each timestamp, for different publisher/subtopic/name combinations.  The default index would just be on the time column and wouldn't allow it.

Reply:
done, indentation fixed.